From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Expose preferences at about:support

Enables the display of certain important preferences at `about:support`.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/toolkit/modules/Troubleshoot.sys.mjs b/toolkit/modules/Troubleshoot.sys.mjs
index dfa6d3a0c816..9d5bf2f3f7e0 100644
--- a/toolkit/modules/Troubleshoot.sys.mjs
+++ b/toolkit/modules/Troubleshoot.sys.mjs
@@ -37,7 +37,10 @@ const PREFS_FOR_DISPLAY = [
   "browser.download.useDownloadDir",
   "browser.fixup.",
   "browser.history_expire_",
+  "browser.ironfox.",
   "browser.link.open_newwindow",
+  "browser.ml.",
+  "browser.phoenix.",
   "browser.places.",
   "browser.privatebrowsing.",
   "browser.search.context.loadInBackground",
@@ -56,6 +59,7 @@ const PREFS_FOR_DISPLAY = [
   "browser.tabs.",
   "browser.theme.",
   "browser.toolbars.",
+  "browser.translations.",
   "browser.urlbar.",
   "browser.zoom.",
   "doh-rollout.",
@@ -65,12 +69,15 @@ const PREFS_FOR_DISPLAY = [
   "extensions.eventPages.enabled",
   "extensions.formautofill.",
   "extensions.lastAppVersion",
+  "extensions.ml.",
   "extensions.quarantinedDomains.enabled",
+  "extensions.update.",
   "extensions.InstallTrigger.enabled",
   "fission.autostart",
   "font.",
   "general.autoScroll",
   "general.useragent.",
+  "geo.",
   "gfx.",
   "html5.",
   "identity.fxaccounts.enabled",
@@ -85,10 +92,12 @@ const PREFS_FOR_DISPLAY = [
   "media.",
   "mousewheel.",
   "network.",
-  "permissions.default.image",
+  "pdfjs.",
+  "permissions.",
   "places.",
   "plugin.",
   "plugins.",
+  "print.enabled",
   "privacy.",
   "security.",
   "services.sync.declinedEngines",
@@ -111,6 +120,7 @@ const PREFS_FOR_DISPLAY = [
   "widget.use-xdg-desktop-portal.mime-handler",
   "widget.gtk.overlay-scrollbars.enabled",
   "widget.wayland",
+  "xpinstall.",
 ];
 
 // The list of prefs we don't display, unlike the list of prefs for display,
@@ -122,6 +132,7 @@ const PREF_REGEXES_NOT_TO_DISPLAY = [
   /^media[.]webrtc[.]debug[.]log_file/,
   /^print[.].*print_to_filename$/,
   /^network[.]proxy[.]/,
+  /^.*NOTE$/,
 ];
 
 // Table of getters for various preference types.
-- 
